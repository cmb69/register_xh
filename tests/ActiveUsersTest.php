<?php

/**
 * Copyright (c) 2013-2023 Christoph M. Becker
 *
 * This file is part of Register_XH.
 */

namespace Register;

use ApprovalTests\Approvals;
use org\bovigo\vfs\vfsStream;
use PHPUnit\Framework\TestCase;
use Register\Infra\ActivityRepository;
use Register\Infra\FakeDbService;
use Register\Infra\FakeRequest;
use Register\Infra\Random;
use Register\Infra\View;

class ActiveUsersTest extends TestCase
{
    private $conf;
    private $activityRepository;
    private $view;

    public function setUp(): void
    {
        vfsStream::setup("root");
        $this->conf = XH_includeVar("./config/config.php", "plugin_cf")["register"];
        $this->activityRepository = new ActivityRepository(
            new FakeDbService("vfs://root/register/active_users.dat", "guest", $this->createMock(Random::class))
        );
        $this->view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["register"]);
    }

    private function sut()
    {
        return new ActiveUsers(
            $this->conf,
            $this->activityRepository,
            $this->view
        );
    }

    public function testRendersActiveUsers(): void
    {
        $this->activityRepository->update("cmb", strtotime("2023-04-16T17:13"));
        $this->activityRepository->update("jane", strtotime("2023-04-16T17:14"));
        $this->activityRepository->update("john", strtotime("2023-04-16T17:15"));
        $response = $this->sut()(new FakeRequest());
        Approvals::verifyHtml($response->output());
    }
}
