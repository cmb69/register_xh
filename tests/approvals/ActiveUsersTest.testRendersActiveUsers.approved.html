<!-- register active users -->
<section class="register_active_users">
  <p>Active Users</p>
  <p>
  <span>cmb</span>
  <span>jane</span>
  <span>john</span>
  </p>
</section>
